### PR TITLE
PWA

### LINKED ISSUE #164: Turn lms into a progressive web application.
- [ ] Add a manifest file.
- [ ] Create a service worker with `start_url`.
- [ ] Add `apple-touch-icon`.

### PR DESCRIPTION
**Pull request by officialpm**
# PWA Support for Python's Course LMS 🚀

 - [x] favicon.ico path fix 🔧
 - [x] Added Icons 🎨
 - [x] Added Manifest.json 🌐 -  Dynamic scope and start_url based on the configuration
 - [x] Added & Updated service worker to 5.1.2 workbox (sw.js) 👷
 - [x] Added routes - /manifest.json, /sw.js ➕

#### fixes #164 🤘🏻

### CODE REVIEW

**Comment by officialpm** [comment]
@yammesicka @gal432 @orronai 
Please can you review this PR

File: lms/templates/base.html [hunk 1]
```diff
@@ -9,6 +9,18 @@
     <link rel="stylesheet" href="{{ url_for('static', filename='my.css') }}">
     <link rel="stylesheet" href="{{ url_for('static', filename='prism.css') }}">
     <link rel="stylesheet" href="{{ url_for('static', filename='hawcons.css') }}">
+    <link rel="manifest" href="{{ url_for('static', filename='manifest.json') }}">
+    <meta name="apple-mobile-web-app-capable" content="yes">
```
Comment (yammesicka) [inline lms/templates/base.html:13]: This is lots of code for just a fallback, I think we should shrink it to be as slim as the W3C example [here](https://www.w3.org/TR/appmanifest/#using-a-link-element-to-link-to-a-manifest).

**Review by yammesicka — CHANGES_REQUESTED** [review]

**Comment by yammesicka** [comment]
Thanks for your great work! I've included some of my ideas in the comments. What do you think?

**Comment by officialpm** [comment]
> Thanks for your great work! I've included some of my ideas in the comments. What do you think?

Yes I will work on them today!! 🤘🏻
It would be great if you include this in hacktoberfest!! Please?

**Comment by yammesicka** [comment]
Sure, every issue merged in this repo is considered as a valid PR towards the Hacktoberfast :)

File: lms/lmsweb/views.py [hunk 2]
```diff
@@ -132,12 +128,27 @@ def logout():
...
 
+@webapp.route('/manifest.json')
+def manifest():
+    return jsonify(MANIFEST)
+
+
+@webapp.route('/sw.js')
+def serviceWorker():
+    response = make_response(send_from_directory(
+        os.path.join(webapp.static_folder),
```
Comment (yammesicka) [inline lms/lmsweb/views.py:145]: Please remove any unnecessary joins :)

File: lms/templates/base.html [hunk 3]
```diff
@@ -9,6 +9,12 @@
...
+    <meta name="apple-mobile-web-app-status-bar-style" content="transparent">
+    <link rel="icon" sizes="16x16 32x32 48x48"  href="{{ url_for('static', filename = 'icons/icons.ico')}}">
```
Comment (yammesicka) [inline lms/templates/base.html:15]: In Python we don't put spaces before and after the equal sign. Please trim them :)

**Review by yammesicka — CHANGES_REQUESTED** [review]

**Comment by yammesicka** [comment]
An excellent job! Just more few simple cleanups to go :)

File: lms/lmsweb/views.py [hunk 4]
```diff
@@ -1,34 +1,29 @@
...
 from functools import wraps
 from typing import Optional
 from urllib.parse import urljoin, urlparse
 
 import arrow  # type: ignore
-from flask import (
-    jsonify, make_response, render_template,
-    request, send_from_directory, url_for,
-)
+from flask import (jsonify, make_response, render_template, request,
```
Comment (yammesicka) [inline lms/lmsweb/views.py:6]: Please keep our sorting conventions in regards to `import`s.

It should be organized from a to z, and split into 3 blocks: standard modules,  3rd party modules and our modules.

File: lms/templates/base.html [hunk 5]
```diff
@@ -9,6 +9,12 @@
     <link rel="stylesheet" href="{{ url_for('static', filename='my.css') }}">
     <link rel="stylesheet" href="{{ url_for('static', filename='prism.css') }}">
     <link rel="stylesheet" href="{{ url_for('static', filename='hawcons.css') }}">
+    <link rel="manifest" href="{{ url_for('static', filename='manifest.json') }}">
+    <meta name="apple-mobile-web-app-capable" content="yes">
+    <meta name="apple-mobile-web-app-status-bar-style" content="transparent">
+    <link rel="icon" sizes="16x16 32x32 48x48"  href="{{ url_for('static', filename='icons/icons.ico')}}">
```
Comment (yammesicka) [inline lms/templates/base.html:15]: Put a space before the `}}` (in this line, and in the line after).

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: lms/lmsweb/views.py [hunk 6]
```diff
@@ -29,6 +28,8 @@
 from lms.utils.consts import RTL_LANGUAGES
 from lms.utils.files import get_language_name_by_extension
 from lms.utils.log import log
+from lms.lmsweb.manifest import MANIFEST
```
Comment (yammesicka) [inline lms/lmsweb/views.py:31]: This import line should appear after `from lms.lmsweb.config import LANGUAGES, LOCALE`

**Review by yammesicka — CHANGES_REQUESTED** [review]

**Review by yammesicka — APPROVED** [review]
Great job! :+1:

**Comment by officialpm** [comment]
> Great job! 👍

Thankyou 💯

**Pull request merged**